### PR TITLE
Improved subscription health problem indicators

### PR DESCRIPTION
**Pull request by piotrrzysko**
- Removed SlowIndicator
- Added parameters to enable/disable indicators
- Fixed indicators to respect delivery type of subscription

### CODE REVIEW

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionMetrics.java [hunk 1]
```diff
@@ -26,7 +27,7 @@ public SubscriptionMetrics(@JsonProperty("delivered") long delivered, @JsonPrope
                                @JsonProperty("otherErrors") String otherErrors, @JsonProperty("codes2xx") String codes2xx,
                                @JsonProperty("codes4xx") String codes4xx, @JsonProperty("codes5xx") String codes5xx,
                                @JsonProperty("Subscription") Subscription.State state, @JsonProperty("rate") String rate,
-                               @JsonProperty("throughput") String throughput) {
+                               @JsonProperty("throughput") String throughput, @JsonProperty("batchRate") String batchRate) {
```
Comment (cristaloleg) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionMetrics.java]: Split into a separate lines please 😉

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/subscription/health/SubscriptionHealthContext.java [hunk 2]
```diff
@@ -1,27 +1,70 @@
...
+    private final double topicRate;
+    private final double subscriptionRate;
+    private final double timeoutsRate;
+    private final double otherErrorsRate;
+    private final double code4xxErrorsRate;
+    private final double code5xxErrorsRate;
+    private final double batchRate;
+    private final long lag;
 
-    public SubscriptionHealthContext(Subscription subscription, TopicMetrics topicMetrics, SubscriptionMetrics subscriptionMetrics) {
```
Comment (cristaloleg) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/subscription/health/SubscriptionHealthContext.java]: IDEA's suggestion? 😄

File: hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionHealthCheckerTest.groovy [hunk 3]
```diff
@@ -1,43 +1,35 @@
...
 import pl.allegro.tech.hermes.api.SubscriptionMetrics
 import pl.allegro.tech.hermes.api.TopicMetrics
 import pl.allegro.tech.hermes.management.domain.subscription.health.SubscriptionHealthChecker
-import pl.allegro.tech.hermes.management.domain.subscription.health.problem.LaggingIndicator
-import pl.allegro.tech.hermes.management.domain.subscription.health.problem.MalfunctioningIndicator
-import pl.allegro.tech.hermes.management.domain.subscription.health.problem.ReceivingMalformedMessagesIndicator
-import pl.allegro.tech.hermes.management.domain.subscription.health.problem.SlowIndicator
-import pl.allegro.tech.hermes.management.domain.subscription.health.problem.TimingOutIndicator
-import pl.allegro.tech.hermes.management.domain.subscription.health.problem.UnreachableIndicator
+import pl.allegro.tech.hermes.management.domain.subscription.health.problem.*
```
Comment (cristaloleg) [inline hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionHealthCheckerTest.groovy]: We don't do this 😉

File: hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionHealthCheckerTest.groovy [hunk 4]
```diff
@@ -1,43 +1,35 @@
...
 import static pl.allegro.tech.hermes.api.SubscriptionHealth.HEALTHY
 import static pl.allegro.tech.hermes.api.SubscriptionHealth.NO_DATA
 import static pl.allegro.tech.hermes.api.SubscriptionHealth.Status.UNHEALTHY
-import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.lagging
-import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.malfunctioning
-import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.receivingMalformedMessages
-import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.slow
-import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.timingOut
-import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.unreachable
+import static pl.allegro.tech.hermes.api.SubscriptionHealthProblem.*
```
Comment (cristaloleg) [inline hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionHealthCheckerTest.groovy]: Same here

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/HealthConfiguration.java [hunk 5]
```diff
@@ -20,31 +20,41 @@
...
-        return new LaggingIndicator(healthProperties.getMaxLagInSeconds());
-    }
-
-    @Bean
-    public SubscriptionHealthProblemIndicator slowIndicator() {
-        return new SlowIndicator(healthProperties.getMinSubscriptionToTopicSpeedRatio());
+        if (healthProperties.isLaggingIndicatorEnabled()) {
+            return new LaggingIndicator(healthProperties.getMaxLagInSeconds());
+        }
+        return new DisabledIndicator();
```
Comment (dankraw) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/HealthConfiguration.java]: can we store a single static instance of the `DisabledIndicator` and return everywhere?

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionMetrics.java (hunk 1 above)
Comment (piotrrzysko) [reply hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionMetrics.java]: done

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/subscription/health/SubscriptionHealthContext.java (hunk 2 above)
Comment (piotrrzysko) [reply hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/subscription/health/SubscriptionHealthContext.java]: yup

File: hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionHealthCheckerTest.groovy (hunk 3 above)
Comment (piotrrzysko) [reply hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/domain/subscription/SubscriptionHealthCheckerTest.groovy]: 👍

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/HealthConfiguration.java (hunk 5 above)
Comment (piotrrzysko) [reply hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/HealthConfiguration.java]: 👍

**Pull request merged**